packages/adblocker/src/engine/engine.ts

```diff
@@ -489,11 +574,31 @@ export default class FilterEngine extends EventEmitter<
   public update({
     newNetworkFilters = [],
     newCosmeticFilters = [],
+    newPreprocessors = [],
     removedCosmeticFilters = [],
     removedNetworkFilters = [],
+    removedPreprocessors = [],
   }: Partial<IListDiff>): boolean {
     let updated: boolean = false;
 
+    // Update preprocessors
+    if (newPreprocessors.length !== 0 || removedPreprocessors.length !== 0) {
+      updated = true;
+      const preservedFilters = this.preprocessors.update({
+        added: newPreprocessors,
+        removed: removedPreprocessors,
+      }).preservedFilters;
+
+      if (removedCosmeticFilters) {
+        removedCosmeticFilters = removedCosmeticFilters.filter(
+          (one) => !preservedFilters.has(one),
```

Comment:
The purpose of having preserveFilters is to prevent filters being removed if there are more than one of preprocessors. If we drop the filter at this stage, the filter won't have a chance to be re-evaluated.

Reply:
I also don't see why we would need `preservedFilters`. Isn't it enough that the PreprocessorBucket keeps track of `excluded`. If not I would prefer to see a unit test to ensure the correctness of this behaviour

Reply:
This is described here: https://ghostery.slack.com/archives/C06JX8NJVMF/p1709280392922439?thread_ts=1708952863.657419&cid=C06JX8NJVMF